Core/Equipment/Service/EquipmentService.cs

```diff
@@ -6,15 +6,53 @@
 using HealthCareInfromationSystem.Core.Equipment.Repository;
 using HealthCareInfromationSystem.Core.Equipment;
 using HealthCareInfromationSystem.repository;
+using HealthCareInfromationSystem.Core.PremiseManagment;
+using System.Data.OleDb;
+using System.Data;
 
 namespace HealthCareInfromationSystem.Core.Equipment.Service
 {
 	class EquipmentService
 	{
 		IEquipmentRepo equipmentRepo = new EquipmentSQL();
         IEquipmentRequestRepo requestRepo = new EquipmentRequestSQL();
+        IPremisseRepo premiseRepo = new PremisseSQL();
 
-		public List<Equipment> LoadEquipmentsFromPremise(string premiseId) {
+        public DataTable SearchByTerm(string term)
+        {
+            return equipmentRepo.SearchByTerm(term);
+        }
+
+        public DataTable SearchByCriteria(string premisesType, string quantity, string equipmentType)
+        {
+            String premisesTypeQuery = premisesType == "" ? "" : $"(pr1.type=\"{premisesType}\" or pr2.type=\"{premisesType}\")";
+
+            String quantityQuery = "";
+            if (quantity == "") quantityQuery = "";
+            else if (quantity == "out of stock") quantityQuery = "eq.quantity=0";
+            else if (quantity == "0-10") quantityQuery = "(eq.quantity > 0 and eq.quantity <= 10)";
+            else if (quantity == "10+") quantityQuery = "eq.quantity > 10";
+
+            String equipmentTypeQuery = equipmentType == "" ? "" : $"eq.type=\"{equipmentType}\"";
+
+            String query = $"" +
```

Comment:
Services should not contain queries, so this may be a problem in the future.

Reply:
Fixed. I moved query building logic from service to repo. Thanks.